src/TestEnvironment.Docker.Containers.Mongo/MongoContainerWaiter.cs

```diff
@@ -17,5 +18,8 @@ protected override async Task<bool> PerformCheck(MongoContainer container, Cance
             await new MongoClient(container.GetConnectionString()).ListDatabasesAsync(cancellationToken);
             return true;
         }
+
+        protected override bool IsRetryable(Exception exception) =>
```

Comment:
And do we need these changes for simple mongo container?

Reply:
`MongoAuthenticationException` means that authentication with Mongo DB is failed and it doesn't depend on Mongo DB health. No need to perform retries in case of authentication failure for standalone Mongo DB mode as well as for replica set mode.